### ISSUE TITLE
Request: `no-extra-parens` option to ignore unary operators

### ISSUE THREAD

**Issue by saveman71** [eslint/eslint#11431]
**What rule do you want to change?** `no-extra-parens`

**Does this change cause the rule to produce more or fewer warnings?** Fewer

**How will the change be implemented? (New option, new default behavior, etc.)?** New option

**Please provide some example code that this change will affect:**

```js
// Note absence of parentheses after the `!`. A quick look might be confusing
// because the `!` could be believed to be part of the RegExp.
if (!/^[A-F0-9-]+$/.test(reference)) {
  this.setState({
    valid: false,
  });
  return;
}

// Here with the extra parens, it looks more like a NOT operator since the parens are more
// "visible" and associated with a "boolean group" if that makes sense.
if (!(/^[A-F0-9-]+$/.test(reference))) {
  this.setState({
    valid: false,
  });
  return;
}
```

**What does the rule currently do for this code?**
It will reject the first example. The new option would allow both.

**What will the rule do after it's changed?**
Allow extra parens to be added so that the behavior of the code is clearer.

**Are you willing to submit a pull request to implement this change?**
Yes

**Comment by saveman71** [comment]
An other example I have is the spread operator.

In my own personal opinion,
```js
dateTime.setHours(...(time.split(':').map(Number)));
```

is clearer than

```js
dateTime.setHours(...time.split(':').map(Number));
```

**Comment by not-an-aardvark** [comment]
Hi, thanks for the proposal.

I'm not sure this option would make sense to add to ESLint core -- it seems like a fairly specific stylistic preference. We wouldn't be able to add individual options for every user's preference, so instead we generally only add an option to a rule if it would be useful for a fairly large percentage of our users, and in other cases we suggest that people [create a custom rule](https://eslint.org/docs/developer-guide/working-with-rules) instead. Personally, I'm not convinced this option would meet the bar for adding to ESLint core, although I wouldn't object to adding it if another team member thinks it's worthwhile.
